### PR TITLE
use URI::db instead of Mojo::URL

### PR DESCRIPTION
**Pull request by Grinnz**
URI::db provides a lot of functionality that is useful for database connections. Along with parsing the URI itself, I used its special methods to simplify determining the database engine to use, parsing out the database name, and generating the connection DSN.

### CODE REVIEW

File: lib/DBIx/TempDB.pm [hunk 1]
```diff
@@ -85,7 +87,7 @@ use constant MAX_OPEN_FDS => eval { use POSIX qw( sysconf _SC_OPEN_MAX ); syscon
 
 our $VERSION = '0.06';
 
-our %SCHEMA_DATABASE = (postgresql => 'postgres', mysql => 'mysql');
+our %SCHEMA_DATABASE = (pg => 'postgres', mysql => 'mysql');
```
Comment (jhthorsen) [inline lib/DBIx/TempDB.pm]: I don't get this change. How can pg-offline.t still be valid after this change?

File: lib/DBIx/TempDB.pm [hunk 2]
```diff
@@ -255,16 +260,19 @@ The default is subject to change!
 
 sub new {
   my $class   = shift;
-  my $url     = Mojo::URL->new(shift || '');
+  my $url     = URI::db->new(shift || '');
+  unless ($url->has_recognized_engine) {
```
Comment (jhthorsen) [inline lib/DBIx/TempDB.pm]: Do we need this? Doesn't the can() below provide the same functionality?

File: lib/DBIx/TempDB.pm [hunk 3]
```diff
@@ -287,7 +295,7 @@ and L<Mojo::mysql>.
 
 =cut
 
-sub url { shift->{url} }
+sub url { shift->{url}->uri }
```
Comment (jhthorsen) [inline lib/DBIx/TempDB.pm]: Why not just return shift->{url} ?

**Comment by jhthorsen** [comment]
There is something I'm missing out on here... It looks like this change makes an URL which is not compatible with Mojo::Pg (which requires the "postgresql" scheme), but I was sure the travis run would fail... Can you please explain to me what I'm missing out on?

File: lib/DBIx/TempDB.pm (hunk 1 above)
Comment (Grinnz) [inline lib/DBIx/TempDB.pm]: The canonical_engine method from the URI::db will always return "pg" for any postgresql scheme.

File: lib/DBIx/TempDB.pm (hunk 2 above)
Comment (Grinnz) [inline lib/DBIx/TempDB.pm]: Perhaps not, I thought it would be a good sanity check

File: lib/DBIx/TempDB.pm (hunk 3 above)
Comment (Grinnz) [inline lib/DBIx/TempDB.pm]: A URI::db created URL is a nested url starting the db: scheme, which will not be usable by Mojo::Pg or Mojo::mysql. The ->uri accessor makes sure that only the inner URL is returned (the one starting with mysql: or pg: etc).

**Comment by Grinnz** [comment]
The URL output by ->url should be the same scheme as was passed to new(), as far as I can tell. There doesn't seem to be a test using Mojo::Pg so I'm not sure what you mean.

**Comment by jhthorsen** [comment]
Awesome @Grinnz! Thank you :+1:

**Pull request merged**